packages/request-manager/src/torControlPort.ts

```diff
@@ -0,0 +1,111 @@
+import net, { Socket } from 'net';
+import fs from 'fs';
+import crypto from 'crypto';
+import util from 'util';
+import path from 'path';
+
+const readFile = util.promisify(fs.readFile);
+const randomBytes = util.promisify(crypto.randomBytes);
+
+const getCookieString = async (authFilePath: string) => {
+    const controlAuthCookiePath = path.join(authFilePath, 'control_auth_cookie');
+    return (await readFile(controlAuthCookiePath)).toString('hex');
+};
+
+const getClientNonce = async () => (await randomBytes(32)).toString('hex');
+
+let clientNonce: string;
+
+interface TorControlPortOptions {
+    host: string;
+    port: number;
+    authFilePath: string;
+}
+
+export class TorControlPort {
+    options: TorControlPortOptions;
+    socket: Socket;
+    isSocketConnected = false;
+
+    constructor(options: TorControlPortOptions) {
+        this.options = options;
+        this.socket = new net.Socket();
+    }
+
+    connect() {
+        return new Promise((resolve, reject) => {
```

Comment:
Can it be an `async/await` like in `TorController`?

Reply:
Probably yes, but I find it way more readable using unwrapped Promise when dealing with event based code. But I will think about it :thinking: